packages/legend-query-builder/src/stores/filter/QueryBuilderFilterStateBuilder.ts

```diff
@@ -70,6 +106,200 @@ const processFilterTree = (
       ),
     );
     filterState.addNodeFromNode(groupNode, parentNode);
+  } else if (
+    matchFunctionName(expression.functionName, [
+      QUERY_BUILDER_SUPPORTED_FUNCTIONS.EXISTS,
+    ])
+  ) {
+    const propertyExpression = guaranteeType(
+      expression.parametersValues[0],
+      AbstractPropertyExpression,
+    );
+    if (
+      propertyExpression.func.value.multiplicity.upperBound === undefined ||
+      propertyExpression.func.value.multiplicity.upperBound > 1
+    ) {
+      const existsNode = new QueryBuilderFilterTreeExistsNodeData(
+        parentFilterNodeId,
+      );
+      const lambdaFunctionInstance = guaranteeType(
+        expression.parametersValues[1],
+        LambdaFunctionInstanceValue,
+        `Can't process filter expression: only supports exists with second paramter as LambdaFunctionInstanceValue`,
+      );
+      const lambdaFunction = guaranteeType(
+        lambdaFunctionInstance.values[0],
+        LambdaFunction,
+      );
+      const filterExpression = guaranteeType(
+        lambdaFunction.expressionSequence[0],
+        SimpleFunctionExpression,
+      );
+      existsNode.setPropertyExpression(propertyExpression);
+      existsNode.lambdaParameterName =
+        lambdaFunction.functionType.parameters[0]?.name;
+      filterState.nodes.set(existsNode.id, existsNode);
+      processFilterTree(filterExpression, filterState, existsNode.id);
+      filterState.addNodeFromNode(existsNode, parentNode);
+    } else {
+      // Build property chain if we have exists filter() and the multiplicity of the property
```

Comment:
are we rewriting the users query here ? 
Seems like we are flattening their exists lambda if they don't have it. 
Not sure if we should do this. The creation of the exists lambda should be flatten as we are guiding users to do this but if the user's query has it in this format i dont think we should override their query.

Reply:
yes we are rewriting user's query here. Seems like it's something we do support in query today. We rewrite the exists for users if the multiplicity of property is less than or equal to 1. Rationale @akphi gave me for this is it is helpful when people change their model multiplicity.

Reply:
will keep it as if for now.